### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by misdaque**
Authentication by Google...

### CODE REVIEW

**Comment by misdaque** [comment]
Closes #43

File: pom.xml [hunk 1]
```diff
@@ -25,6 +25,33 @@
 
     <dependencies>
 
+        <dependency>
+            <groupId>org.springframework.security.oauth.boot</groupId>
+            <artifactId>spring-security-oauth2-autoconfigure</artifactId>
+            <version>2.2.2.RELEASE</version>
```
Comment (keensam04) [inline pom.xml]: remove version

File: pom.xml [hunk 2]
```diff
@@ -25,6 +25,33 @@
...
+        </dependency>
+
+<!--        <dependency>
```
Comment (keensam04) [inline pom.xml]: remove commented dependency

File: pom.xml [hunk 3]
```diff
@@ -25,6 +25,33 @@
...
+
+        <dependency>
+            <groupId>org.springframework.security</groupId>
+            <artifactId>spring-security-oauth2-jose</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>io.jsonwebtoken</groupId>
+            <artifactId>jjwt</artifactId>
+            <version>0.9.0</version>
```
Comment (keensam04) [inline pom.xml]: move version to properties

File: src/main/java/com/autohome/Main.java [hunk 4]
```diff
@@ -18,4 +19,17 @@ public static void main(String[] args) {
     private static void logShutDown() {
         LOG.info("JVM shutting down");
     }
+
```
Comment (keensam04) [inline src/main/java/com/autohome/Main.java:22]: remove commented code

File: src/main/java/com/autohome/config/SecurityConfig.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+@EnableWebSecurity(debug = true)
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    private CustomUserOidcService userOidcService;
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+
+       /* http.authorizeRequests()
```
Comment (keensam04) [inline src/main/java/com/autohome/config/SecurityConfig.java]: remove commented code

File: src/main/java/com/autohome/service/UserService.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserService implements UserDetailsService {
+
+/*    @Override
```
Comment (keensam04) [inline src/main/java/com/autohome/service/UserService.java]: remove commented code

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -6,3 +6,16 @@ logging.file=autohome.logs
 spring.datasource.url=jdbc:mysql://192.168.1.250:3306/autohome
 spring.datasource.driverclassname = com.mysql.cj.jdbc.Driver
 
+spring.security.oauth2.client.registration.google.clientId=750505933588-ns93dn1hlkfj54bu6e7eg4p9m8hckb40.apps.googleusercontent.com
+spring.security.oauth2.client.registration.google.clientSecret=pRZrbv64X4WtZHhQo30sZHWd
+#security.oauth2.client.project_id=zinc-prism-275020
```
Comment (keensam04) [inline src/main/resources/application.properties]: remove commented block

File: src/main/resources/application.properties [hunk 8]
```diff
@@ -6,3 +6,16 @@ logging.file=autohome.logs
 spring.datasource.url=jdbc:mysql://192.168.1.250:3306/autohome
 spring.datasource.driverclassname = com.mysql.cj.jdbc.Driver
 
+spring.security.oauth2.client.registration.google.clientId=750505933588-ns93dn1hlkfj54bu6e7eg4p9m8hckb40.apps.googleusercontent.com
```
Comment (keensam04) [inline src/main/resources/application.properties]: remove credentials from prop file. to be configured on the server

**Review by keensam04 — CHANGES_REQUESTED** [review]

**Comment by keensam04** [comment]
Deprecated by #44

**Pull request closed without merging**